pyte/screens.py

```diff
@@ -592,11 +605,13 @@ def insert_lines(self, count=None):
 
         # If cursor is outside scrolling margins it -- do nothin'.
         if top <= self.cursor.y <= bottom:
-            #                           v +1, because range() is exclusive.
-            for line in range(self.cursor.y,
-                              min(bottom + 1, self.cursor.y + count)):
-                self.buffer.pop(bottom)
-                self.buffer.insert(line, take(self.columns, self.default_line))
+            # Shift old lines down (backwards)
+            for old_line in range(bottom, self.cursor.y - 1, -1):
+                if old_line + count <= bottom:  # There is place inside margins
```

Comment:
Because we need to process all these lines anyway. Yes, in some cases lines will be rewriten by moved lines, but consider inserting lines to the bottom. We need just popping then.

After all, "scanline" cycle is easier to understand, than cycle passing some lines.